### ISSUE TITLE
Support for SwiftLint in pre commit

### ISSUE THREAD

**Issue by kpavankotesh** [brigade/overcommit#473]
Hey, 

I would like to use this for iOS projects as well and we use SwiftLint. Seems like you don't support SwiftLint in the pre-commit section. Can someone guide me on adding support to SwiftLint?

**Comment by sds** [comment]
Hey @kpavankotesh,

Thanks for your interest in adding a `SwiftLint` pre-commit hook to Overcommit!

You can see a recent example of adding a simple linter by looking at the pull request which added the `TSLint` pre-commit hook: https://github.com/brigade/overcommit/pull/421/files

**Comment by Ashraf-Ali-aa** [comment]
@kpavankotesh @sds Something like this would work
```
module Overcommit::Hook::PreCommit
  # Runs `swiftlint lint` against any modified Swift files.
  #
  # @see https://github.com/realm/SwiftLint
  class SwiftLint < Base
    MESSAGE_REGEX = /^(?<file>(?:\w:)?[^:]+):(?<line>\d+)[^ ]* (?<type>[^ ]+):(?<message>.*):/

    MESSAGE_TYPE = lambda do |type|
      type.include?('warning') ? :warning : :error
    end

    def run
      result = execute(command, args: applicable_files)
      output = result.stdout.split("\n").map { |message| [message.match(MESSAGE_REGEX)].join.chomp(':') }

      return :pass if result.success?

      # example message:
      #   path/to/file.swift:1: Error type: Error message: details
      extract_messages(
        output,
        MESSAGE_REGEX,
        MESSAGE_TYPE
      )
    end
  end
end
```


```
  SwiftLint:
    enabled: true
    description: 'Analyze with swift-lint'
    required_executable: 'swiftlint'
    flags: ['lint', '--path']
    include: '**/*.swift'
```

**Comment by kpavankotesh** [comment]
Okay. I'll check this and create a pull request on this. Thanks for the directions.

**Comment by kpavankotesh** [comment]
https://github.com/brigade/overcommit/pull/474 any help?

**Comment by sds** [comment]
Implemented in #603
